### ISSUE TITLE
authentication triggers an error simplejson.errors.JSONDecodeError

### ISSUE THREAD

**Issue by burmkiller** [vvaezian/metabase_api_python#12]
I installed the package and I tried to authenticate with both the command line and a script, yet it's not working. When I run this code

```
from metabase_api import Metabase_API

Metabase_API("http://localhost:4009/","test@test.com","test123")

```

I'm getting the following error in the my terminal

```
Traceback (most recent call last):
  File "mbmap.py", line 59, in <module>
    Metabase_API("http://localhost:4009/","test@test.com","test123")
  File "/home/test/.local/lib/python3.8/site-packages/metabase_api/metabase_api.py", line 13, in __init__
    self.authenticate()
  File "/home/test/.local/lib/python3.8/site-packages/metabase_api/metabase_api.py", line 25, in authenticate
    self.session_id = res.json()['id']
  File "/usr/lib/python3/dist-packages/requests/models.py", line 897, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3/dist-packages/simplejson/__init__.py", line 518, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

```

**Comment by vvaezian** [comment]
Hi @burmkiller,
First make sure you can connect to `http://localhost:4009/` using a browser.

If the previous step is successful, then use the following code to debug the issue. I have added a line for printing the data.
```python
import requests
import getpass

class Metabase_API():
  
  def __init__(self, domain, email, password=None):
    
    self.domain = domain
    self.email = email
    self.password = getpass.getpass(prompt='Please enter your password: ') if password is None else password
    self.session_id = None
    self.header = None
    self.authenticate()
  
  
  def authenticate(self):
    """Get a Session ID"""
    conn_header = {'username':self.email,
                   'password':self.password}

    res = requests.post(self.domain + '/api/session', json = conn_header)
    
    # printing the returned result
    print(res.status_code, res.text)
    
    if not res.ok:
      raise Exception(res)
    
    self.session_id = res.json()['id']
    self.header = {'X-Metabase-Session':self.session_id}


mb = Metabase_API('http...', 'user', 'pass')
```

**Comment by vvaezian** [comment]
@burmkiller was your issue resolved?

**Comment by brianspurling** [comment]
@vvaezian I just got this same issue (first time trying to use the package).

The printout from your test code gives me:

```
200
```

Note `res.text` is an empty string.  `print(type(res.text))` returns `<class 'str'>`

**Comment by vvaezian** [comment]
@brianspurling Please try with a newer version of Metabase, and let me know. It may have been the case that in that version the API behavior was different.
If the problem persists, please provide error logs.

**Comment by brianspurling** [comment]
Thannks @vvaezian . 

We just got our Metabase upgraded to v0.38.1

![image](https://user-images.githubusercontent.com/13377821/112469512-7ccbb080-8d61-11eb-8de3-17a1859dfc74.png)

Unfortunately I get exactly the same from your test code: 200, and `res.text` is an empty string :(

**Comment by vvaezian** [comment]
@brianspurling Are you using a deployed version, or just a local JAR file for running Metabase?
If local Jar, make sure the port number is correct. By default it is 3000, but the person who created this ticket was using port 4009 so I mentioned that in my comment (`http://localhost:4009/`).
If you are using a deployed version, which method is it (ElasticBeanstalk, Docker on EC2, ...)?

**Comment by brianspurling** [comment]
It's deployed. I think it’s just deployed directly onto EC2. It _might_ be docker. Is that significant?

**Comment by vvaezian** [comment]
What are the Metabase logs (Admin panel > Troubleshooting > Logs) when you try to connect using the Python package and get the error?

**Comment by brianspurling** [comment]
Nothing, unfortunately. And to be clear, I'm getting a 200 back from the request - not an error response. The error is caused by the response body being empty. 

Not sure if this is relevant, but it returns _really_ quickly. It doesn't feel like it's doing anything on the remote. But obviously I have no way of knowing that!

**Comment by aattpcfish** [comment]
It might be the domain . 
input `http://localhost:4009/` 
`res = requests.post(self.domain + '/api/session', json = conn_header)`

and call `http://localhost:4009//api/session`  lead to the result with r`''`, which status code is still 200.

**Comment by ericknudson** [comment]
Just ran into the same issue - getting a JSONDecodeError and a 200 back from the request (metabase v0.39.2). Using the deployed version and I don't see any relevant logs - any advice for how else to troubleshoot?

**Comment by vvaezian** [comment]
@ericknudson I cannot reproduce the error, so it is not straightforward to fix this issue. Apparently a small portion of users have this issue, but not sure what is causing this.

**Comment by chlebovec** [comment]
I had same issue and it was exactly what @aattpcfish pointed, removal of last `/` solved it. Perhaps you may use `os.path.join(self.domain, 'api', 'session')` instead of `self.domain + '/api/session'`

**Comment by vvaezian** [comment]
@chlebovec I see.
I'll change the code so it works fine no matter whether the user enters the `/` at the end or not.

**Comment by vvaezian** [comment]
@burmkiller @brianspurling @aattpcfish @ericknudson please try connecting without entering the `/` at the end of domain, and see if still there is any issue.

**Comment by vvaezian** [comment]
Fixed in 0.2.12

**Comment by vvaezian** [comment]
@burmkiller @brianspurling @aattpcfish @ericknudson @chlebovec
Hi folks,
At Intellimenta we have developed [DataBot](https://intellimenta.com/products/databot/), a conversational analytics app that connects with **Metabase** and enables users to ask questions about data and visualize the results using natural language in their preferred language. 
If your company uses Metabase, DataBot can significantly enhance how your team and customers interact with data. If you're interested in learning more, you can book a demo using [this](https://outlook.office365.com/owa/calendar/booking@intellimenta.com/bookings/s/ZEMZuuM07U6C4Tr-GKUyOQ2) link. Happy to discuss further.
